test/y2partitioner/dialogs/lvm_vg_test.rb

```diff
@@ -0,0 +1,542 @@
+  end
+
+  describe Y2Partitioner::Dialogs::LvmVg::NameWidget do
+    subject { described_class.new(controller) }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#init" do
+      before do
+        controller.vg_name = "vg1"
+      end
+
+      it "sets its value with vg name stored in the controller" do
+        expect(subject).to receive(:value=).with("vg1")
+        subject.init
+      end
+
+      it "gets focus" do
+        expect(Yast::UI).to receive(:SetFocus).with(Id(subject.widget_id))
+        subject.init
+      end
+    end
+
+    describe "#handle" do
+      before do
+        allow(subject).to receive(:value).and_return("vg1")
+      end
+
+      it "stores in the controller the given vg name" do
+        subject.handle
+        expect(controller.vg_name).to eq("vg1")
+      end
+
+      it "returns nil" do
+        expect(subject.handle).to be_nil
+      end
+    end
+
+    describe "#validate" do
+      before do
+        controller.vg_name = vg_name
+      end
+
+      let(:vg_name) { nil }
+
+      it "gets focus" do
+        expect(Yast::UI).to receive(:SetFocus).with(Id(subject.widget_id))
+        subject.init
+      end
+
+      context "when the vg name is not given" do
+        let(:vg_name) { "" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          subject.validate
+        end
+
+        it "returns false" do
+          expect(subject.validate).to eq(false)
+        end
+      end
+
+      context "when the vg name has invalid characters" do
+        let(:vg_name) { "vg%" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          subject.validate
+        end
+
+        it "returns false" do
+          expect(subject.validate).to eq(false)
+        end
+      end
+
+      context "when a device exists with the given vg name" do
+        let(:vg_name) { "sda" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          subject.validate
+        end
+
+        it "returns false" do
+          expect(subject.validate).to eq(false)
+        end
+      end
+
+      context "when the given vg name is valid" do
+        let(:vg_name) { "vg100" }
+
+        it "does not show an error popup" do
+          expect(Yast::Popup).to_not receive(:Error)
+          subject.validate
+        end
+
+        it "returns true" do
+          expect(subject.validate).to eq(true)
+        end
+      end
+    end
+  end
+
+  describe Y2Partitioner::Dialogs::LvmVg::ExtentSizeWidget do
+    subject { described_class.new(controller) }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#init" do
+      before do
+        controller.extent_size = "128 KiB"
+      end
+
+      it "sets its value with the extent size stored in the controller" do
+        expect(subject).to receive(:value=).with("128 KiB")
+        subject.init
+      end
+    end
+
+    describe "#handle" do
+      before do
```

Comment:
I would like to see a unit test for empty string and/or an invalid string here. Just to ensure it doesn't break today or in the future.

Reply:
Added.